cloudinit/sources/DataSourceMAAS.py

```diff
@@ -40,6 +43,12 @@ class DataSourceMAAS(sources.DataSource):
     dsname = "MAAS"
     id_hash = None
     _oauth_helper = None
+    network_config_sources: Tuple[NetworkConfigSource, ...] = (
```

Comment:
Did this need to be re-ordered in the non-local case? Why does initramfs need to take priority over kernel cmdline?

Reply:
I agree here, we don't want to change network config source ordering on the non-local DataSourceMAAS case. We would prefer to avoid that change in behavior and limit the scope of this PR to just what happens in init-local when initramfs network config is present.


From the related MAAS Jira card linked:
` In the case where the system boots from disk, initramfs doesn’t bring up the networking`.  
This means that we should only end up detecting DataSourceMAASLocal during diskless ephemeral boots on MAAS and we don't want to alter any other behavior for this datasources beyond that.

So, I suggest drop this section of the changeset.

Reply:
Oops, I botched merging in the changes from the patch, sorry for the confusion here.